packages/odyssey-react-mui/src/Select.tsx

```diff
@@ -208,6 +255,25 @@ const Select = <
     [onChangeProp]
   );
 
+  const removeSelection = useCallback(
+    (itemToRemove: string) => {
+      if (!Array.isArray(internalSelectedValues)) {
+        return;
+      }
+
+      const newValue = internalSelectedValues!.filter(
+        (item: string) => item !== itemToRemove
+      );
```

Comment:
I don't like using `!` in `internalSelectedValues!` if we don't need to. Is there a reason TS isn't seeing that we have an array at this point?

Instead of doing a negative check at the top (early fail), can we do an explicit array check and fall into this `newValue` creation only if it's an array?

Like so:

```suggestion
      if (Array.isArray(internalSelectedValues)) {
         const newValue = internalSelectedValues!.filter(
           (item: string) => item !== itemToRemove
         );
         
         // ....
      }
```